dev/com.ibm.ws.ejbcontainer.singleton.startup_fat/test-applications/SingletonPropertyTest.war/src/singleton/property/web/AllowWorkNoStartupServlet.java

```diff
@@ -0,0 +1,110 @@
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.servlet.annotation.WebServlet;
+
+import componenttest.app.FATServlet;
+import singleton.property.shared.StartupHelperSingleton;
+import singleton.property.shared.TestData;
+
+/**
+ * Test the behavior of the com.ibm.websphere.ejbcontainer.blockWorkUntilAppStartedWaitTime and
+ * com.ibm.websphere.ejbcontainer.blockWorkUntilAppStarted properties for an application that does
+ * not contain any Startup Singleton beans.
+ *
+ * During application startup of an application that does not contain any Startup Singleton bean, by
+ * default, outside work will be blocked. However, if the "blockWorkUnitlAppStarted" property is set
+ * to false then outside work will not be blocked
+ *
+ * The following property values have been set in jvm.options for the server:
+ * -Dcom.ibm.websphere.ejbcontainer.blockWorkUntilAppStartedWaitTime=20
+ * -Dcom.ibm.websphere.ejbcontainer.blockWorkUntilAppStarted=false
+ */
+@SuppressWarnings("serial")
+@WebServlet("/AllowWorkNoStartupServlet")
+public class AllowWorkNoStartupServlet extends FATServlet {
+    private final static String CLASSNAME = AllowWorkNoStartupServlet.class.getName();
+    private final static Logger svLogger = Logger.getLogger(CLASSNAME);
+
+    private static final String JNDI_STARTUP_HELPER = "java:global/SingletonPropertyNoStartup/SingletonPropertyNoStartupBean/NoStartupHelperSingletonBean";
+
+    private StartupHelperSingleton lookupHelper() throws NamingException {
+        svLogger.info("lookupHelper : jndi name = " + JNDI_STARTUP_HELPER);
+        StartupHelperSingleton helper = (StartupHelperSingleton) new InitialContext().lookup(JNDI_STARTUP_HELPER);
+        svLogger.info("lookupHelper : returning : " + ((helper == null) ? helper : helper.getClass().getName()));
+        return helper;
+    }
+
+    /**
+     * This test verifies 2 things. First it will verify, with the blockWorkUntilAppStarted property
+     * set to false (disabled), that work will not be blocked since the application does not contain
+     * Startup Singletons. And second, it will verify that the postConstruct for non-startup Singleton,
+     * with the blockWorkUntilAppStartedWaitTime set to 20 seconds, that the work beans will not run
+     * until the bean is first used.
+     *
+     * To accomplish this the application has a ServletContextListener that uses a
+     * java.util.concurrent.CyclicBarrier that allows the processing of the SCL to wait for our test
+     * thread to signal it to continue. This allows us to have the application startup processing to
+     * be held indefinitely, which allows the test to attempt bean access before startup completes.
+     */
+    // @Test - TODO : com.ibm.websphere.ejbcontainer.blockWorkUntilAppStarted not supported on Liberty
```

Comment:
```suggestion
    @Test
    @Ignore("com.ibm.websphere.ejbcontainer.blockWorkUntilAppStarted not supported on Liberty") //TODO
```

Nitpick: I prefer to keep the test annotation, and add in the @Ignore annotation for two reasons. 1. you can verify now that your test will be included when it does get enabled. 2. the reason it was ignored will be output into the test results which helps when there are so many other filtering methods out there.